src/commands/scanner/rule/list.ts

```diff
@@ -59,58 +61,71 @@ export default class List extends SfdxCommand {
+  private async getPmdRules() : Promise<AnyJson> {
+    // PmdCatalogWrapper is a layer of abstraction between the plugin and PMD, facilitating code reuse and other goodness.
+    return new PmdCatalogWrapper().getCatalog()
+      .then((catalog : {rules}) => {
+        // The catalog has a bunch of information on it, but we only need the 'rules' property.
+        const rules = catalog.rules;
+        // We'll want to do some light tampering with the JSON in order to make sure it displays cleanly on the list.
+        return rules.map((rule : {rulesets : string[]}) => {
+          // If any of the rulesets have a name longer than 20 characters, we'll truncate it to 15 and append ellipses,
+          // so it doesn't overflow horizontally.
+          let truncatedRulesets = rule.rulesets.map((ruleset : string) => {
+            return ruleset.length >= 20 ? ruleset.slice(0, 15) + '...' : ruleset;
+          });
+          const cleanedRule = JSON.parse(JSON.stringify(rule));
+          cleanedRule.rulesets = truncatedRulesets;
+          return cleanedRule;
+        });
+      });
+  }
 
-    // Since loading the rules might take a while, log something at the start so the user doesn't think we're hanging.
-    this.ux.log(messages.getMessage('outputTemplates.preparing'));
+  private ruleSatisfiesFilterConstraints(rule : {categories; rulesets; languages}) : boolean {
+    // Get the filter criteria from the input flags.
+    const filterCats = this.flags.category || [];
+    const filterRulesets =  this.flags.ruleset || [];
+    const filterLangs = this.flags.language || [];
 
-    return this.getRules(cats, rulesets, sev, langs, author)
-      .then((res : AnyJson[]) => {
-        this.ux.table(res, ['name', 'categories', 'rulesets', 'languages', 'author']);
-        // This JSON is displayed when the --json flag is provided.
-        // TODO: The shape of this JSON will need to change.
-        return res;
-      }, (rej : string) => {
-        return {};
-      });
+    // If the user specified one or more categories, this rule must be a member of at least one of those categories.
+    if (filterCats.length > 0 && !this.listContentsOverlap(filterCats, rule.categories)) {
+      return false;
+    }
+
+    // If the user specified one or more rulesets, this rule must be a member of at least one of those rulesets.
+    if (filterRulesets.length > 0 && !this.listContentsOverlap(filterRulesets, rule.rulesets)) {
+      return false;
+    }
+
+    // If the user specified one or more languages, this rule must apply to at least one of those languages.
+    if (filterLangs.length > 0 && ! this.listContentsOverlap(filterLangs, rule.languages)) {
+      return false;
+    }
+
+    // If we didn't find any reasons to disqualify the rule, it's good.
+    return true;
+  }
+
+  private listContentsOverlap(list1 : string[], list2 : string[]) : boolean {
```

Comment:
This method can simplify down to just:

`list1.find(x => list2.includes(x));`

If you wanted the actual items that intersected, can use

`let intersection = list1.filter(x => list2.includes(x));`